precompiles/outposts/osmosis/utils_test.go

```diff
@@ -0,0 +1,445 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package osmosis_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	erc20types "github.com/evmos/evmos/v15/x/erc20/types"
+
+	"github.com/evmos/evmos/v15/precompiles/outposts/osmosis"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/crypto/tmhash"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	tmtypes "github.com/cometbft/cometbft/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	"github.com/cosmos/cosmos-sdk/testutil/mock"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	evmosapp "github.com/evmos/evmos/v15/app"
+	evmosibc "github.com/evmos/evmos/v15/ibc/testing"
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
+	evmosutil "github.com/evmos/evmos/v15/testutil"
+	evmosutiltx "github.com/evmos/evmos/v15/testutil/tx"
+	evmostypes "github.com/evmos/evmos/v15/types"
+	"github.com/evmos/evmos/v15/utils"
+	"github.com/evmos/evmos/v15/x/evm/statedb"
+	evmtypes "github.com/evmos/evmos/v15/x/evm/types"
+	feemarkettypes "github.com/evmos/evmos/v15/x/feemarket/types"
+	inflationtypes "github.com/evmos/evmos/v15/x/inflation/types"
+)
+
+const (
+	portID    = "transfer"
+	channelID = "channel-0"
+)
+
+const (
+	TokenToMint = 9e18
+)
+
+// SetupWithGenesisValSet initializes a new EvmosApp with a validator set and genesis accounts
+// that also act as delegators. For simplicity, each validator is bonded with a delegation
+// of one consensus engine unit (10^6) in the default token of the simapp from first genesis
+// account. A Nop logger is set in SimApp.
+func (s *PrecompileTestSuite) SetupWithGenesisValSet(
```

Comment:
we don't need this anymore. We can use the new UnitIntegration test struct that Fraddy enabled